packages/adapter-node/src/NodeCanvasElement.ts

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable func-names */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+
+import { Canvas, CanvasRenderingContext2D, Image, NodeCanvasRenderingContext2DSettings } from 'canvas';
+import EventEmitter from 'events';
+import createGLContext from 'gl';
+import { ContextIds } from 'pixi.js';
+
+const _ctx = Symbol('ctx');
+
+function putImageData(gl: WebGLRenderingContext, canvas: NodeCanvasElement)
+{
+    const { width, height } = canvas;
+
+    const ctx = canvas[_ctx] as CanvasRenderingContext2D;
+
+    const data = ctx.getImageData(0, 0, width, height);
+
+    const pixels = new Uint8Array(width * height * 4);
+
+    gl.readPixels(0, 0, width, height, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+
+    for (let i = 0; i < height; i++)
+    {
+        for (let j = 0; j < width; j++)
+        {
+            const col = j;
+            const row = height - i - 1;
+
+            for (let k = 0; k < 4; k++)
+            {
+                const idx = (4 * ((row * width) + col)) + k;
+                const idx2 = (4 * ((i * width) + col)) + k;
+
+                data.data[idx] = pixels[idx2];
+            }
+        }
+    }
+
+    ctx.putImageData(data, 0, 0);
+
+    return ctx;
+}
+export class NodeCanvasElement extends Canvas
+{
+    private __event__: EventEmitter;
+    private __gl__: WebGLRenderingContext;
+    private __contextType__: ContextIds;
```

Comment:
We don't really have this naming pattern anywhere in Pixi. Would suggest more simple `_event`, `_gl` and `_contextType`. Unless there's a reason for doing this?